src/ui/OwlUI.ts

```diff
@@ -47,28 +53,56 @@ export class OwlUI {
         };
     }
     /**
-     * Mount all of a plugin's Components
+     * Mount all of a plugin's Components to Owl and register its components'
+     * intents, actions and commands (TODO: this does not work for
+     * sub-components since those are mounted by their parent and we do not have
+     * access to their instantiation).
      *
      * @param {JWOwlUIPlugin} pluginInstance
      * @returns {Promise<void>}
      */
-    async _mountPluginComponents(pluginInstance: JWOwlUIPlugin): Promise<void> {
+    async _handleComponents(pluginInstance: JWOwlUIPlugin): Promise<void> {
```

Comment:
Same remark about instances being the default case. I'd rename the parameter to `plugin`.